staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -416,13 +348,28 @@ func (p *smpPatcher) firstPatchAttempt(currentObject runtime.Object, currentReso
 }
 
 func (p *smpPatcher) subsequentPatchAttempt(currentObject runtime.Object, currentResourceVersion string) (runtime.Object, error) {
+	return subsequentPatchLogic(p.patcher, p, currentObject, currentResourceVersion)
+}
+
+type patchSource interface {
+	originalStrategicMergePatch() (map[string]interface{}, error)
+	computeStrategicMergePatch(unversionedObject runtime.Object, currentVersionedObjMap map[string]interface{}) (map[string]interface{}, error)
+}
+
+func subsequentPatchLogic(p *patcher, ps patchSource, currentObject runtime.Object, currentResourceVersion string) (runtime.Object, error) {
 	// on a conflict (which is the only reason to have more than one attempt),
 	// 1. build a strategic merge patch from originalJS and the patchedJS.  Different patch types can
 	//    be specified, but a strategic merge patch should be expressive enough handle them.  Build the
 	//    patch with this type to handle those cases.
 	// 2. build a strategic merge patch from originalJS and the currentJS
 	// 3. ensure no conflicts between the two patches
 	// 4. apply the #1 patch to the currentJS object
+	//
+	// TODO: Instead of computing two 2-way merges and comparing them for
```

Comment:
Rather than this TODO, I'm not actually convinced we should be treating subsequent attempts any differently than the first attempt. Unless a resourceVersion is provided in the patch (in which case a conflict should probably immediately fail with no retry), the delta between the object in etcd from the first attempt to the present isn't particularly meaningful. Worth more discussion, but I think I'd be in favor of dropping subsequentPatchLogic entirely

Reply:
That would be a behavior change and this PR is supposed to be a total no-op.

This behavior only makes sense if the user specified an RV. I actually think it is so unnecessary that it's a backwards compatible change to drop this behavior if the user didn't specify an RV (!). But not in this PR. I can add this as an additional TODO.

We can additionally argue about whether it makes sense if the RV is specified.

Reply:
> That would be a behavior change and this PR is supposed to be a total no-op.

Oh, definitely a follow-up, I appreciate this PR changing no behavior.